### PR TITLE
Calculator App

### PR DESCRIPTION
**Pull request by keawade**
A PR to track progress for my calculator app.

### CODE REVIEW

**Comment by keawade** [comment]
Base functionality is complete!

**Comment by NathanBland** [comment]
@keawade, you should use `@click` instead of `v-on`. It works either way but using `@click` keeps in consistent with the rest of the system.

**Comment by keawade** [comment]
- [ ] Add additional buttons under an advanced controls panel
- [x] Add keyboard control
- [ ] Add history

File: package.json [hunk 1]
```diff
@@ -13,8 +13,9 @@
     "lint": "eslint --ext .js,.vue src test/unit/specs test/e2e/specs"
   },
   "dependencies": {
-    "vue": "^1.0.21",
-    "babel-runtime": "^6.0.0"
+    "babel-runtime": "^6.0.0",
+    "mathjs": "^3.3.0",
```
Comment (NathanBland) [inline package.json]: This should be a `devDependency`, as a production build will consolidate all of those into one file. No need for this to be a primary dependency.

**Comment by NathanBland** [comment]
@keawade Looking forward to keyboard bindings!

For your first objective, you could probably just build a second component, and put both under one folder.. Then if you use `<component :is='<mode>'></component>` you would be able to toggle between them rather effortlessly

**Comment by keawade** [comment]
Does w-os have window focus yet? Something like that will be needed to allow the right window to capture key strokes.

**Comment by NathanBland** [comment]
Actually, with the way events are captured, aka `@keypress` they are handled on a per component basis _unless_ they are explicitly placed into the overall app structure. The only thing that does this currently is a click monitor for the `app` menu

**Comment by NathanBland** [comment]
> Does w-os have window focus yet? Something like that will be needed to allow the right window to capture key strokes.

I was wrong about this I think.. I'm going to try and add a `hasFocus` to the `wApp` wrapper..

**Comment by NathanBland** [comment]
As of commit [aed9b60](https://github.com/NathanBland/w-os/commit/aed9b60bab7bb47f8906c19f3d084380ff7fd110) each `wApp` get's a `tabindex=0` property, which allows focus. Also, each `wApp` wrapper has a `hasFocus` property that is triggered when they are the top level element. See the app [`terminal`](https://github.com/NathanBland/w-os/blob/ceb54dc5c932e20aac260ecaeae00c89a821d444/src/components/apps/terminal/terminal.vue) for example usage.

@keawade Do you consider this stable enough to merge, or would you rather wait for button bindings?

**Comment by keawade** [comment]
@NathanBland It's currently stable, but I'd argue the button bindings are core functionality. I think we should wait to merge it.

**Comment by keawade** [comment]
@NathanBland I suppose we could merge it so @crodeheaver will be happy.

**Comment by crodeheaver** [comment]
+1

**Comment by keawade** [comment]
@NathanBland With keyboard controls implemented, I think we can pull this into the main repo. :tada:

@crodeheaver (I hope you're happy :P)

**Comment by NathanBland** [comment]
I didn't even code review of your latest stuff, so hopefully it's all ok

**Pull request merged**